settings.gradle.kts

```diff
@@ -195,3 +195,25 @@ gradle.settingsEvaluated {
         throw GradleException("This build requires JDK 11. It's currently ${getBuildJavaHome()}. You can ignore this check by passing '-Dorg.gradle.ignoreBuildJavaVersionCheck=true'.")
     }
 }
+
+// region platform include DSL
+
+fun includeSubproject(projectName: String) =
+    includeAt(projectName, file("subprojects/$projectName"))
+
+fun platform(platformName: String, platformConfiguration: PlatformScope.() -> Unit) =
+    PlatformScope(platformName).platformConfiguration()
+
+class PlatformScope(
+    private val platformName: String
+) {
+    fun subproject(projectName: String) =
```

Comment:
💭 I like this little DSL a lot.  Should this be named `includeSubproject` as well, for consistency?  Probably ought to be defined in a plugin, to follow our own guidelines.

Reply:
I agree it should be defined in a plugin, but it seems `build-logic-settings` has been removed. @gradle/bt-developer-productivity What do you advise we do here? Re-create `build-logic-settings`?

Reply:
I'd prefer to not re-create build-logic settings if we can avoid it. It caused some headaches while we had it, especially since it doesn't inherit the build cache configuration.

Reply:
Sounds good. We can leave this as-is